src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py

```diff
@@ -342,858 +344,859 @@ def _encode_target(data, target, mapping=None):
-            - 'log_loss'         : Log loss
-            - 'confusion_matrix' : An SFrame with counts of possible
-                                   prediction/true label combinations.
-            - 'roc_curve'        : An SFrame containing information needed for an
-                                   ROC curve
-
-        Returns
-        -------
-        out : dict
-            Dictionary of evaluation results where the key is the name of the
-            evaluation metric (e.g. `accuracy`) and the value is the evaluation
-            score.
-
-        See Also
-        ----------
-        create, predict
-
-        Examples
-        ----------
-        .. sourcecode:: python
-
-          >>> results = model.evaluate(data)
-          >>> print results['accuracy']
-        """
-        return self.__proxy__.evaluate(dataset, metric)
-
-
-class ActivityClassifier(_CustomModel):
-    """
-    A trained model that is ready to use for classification or export to
-    CoreML.
-
-    This model should not be constructed directly.
-    """
-
-    _PYTHON_ACTIVITY_CLASSIFIER_VERSION = 2
-
-    def __init__(self, state):
-        self.__proxy__ = _PythonProxy(state)
-
-    def _get_native_state(self):
-        from .._mxnet import _mxnet_utils
-        state = self.__proxy__.get_state()
-        state['_pred_model'] = _mxnet_utils.get_mxnet_state(state['_pred_model'])
-        return state
-
-    @classmethod
-    def _load_version(cls, state, version):
-        from ._mx_model_architecture import _define_model_mxnet
-        from .._mxnet import _mxnet_utils
-
-        _tkutl._model_version_check(version, cls._PYTHON_ACTIVITY_CLASSIFIER_VERSION)
-
-        data_seq_len = state['prediction_window'] * state['_predictions_in_chunk']
-
-        context = _mxnet_utils.get_mxnet_context(max_devices=state['num_sessions'])
-        _, _pred_model = _define_model_mxnet(len(state['_target_id_map']), state['prediction_window'],
-                                             state['_predictions_in_chunk'], context)
-
-        batch_size = state['batch_size']
-        preds_in_chunk = state['_predictions_in_chunk']
-        win = state['prediction_window'] * preds_in_chunk
-        num_features = len(state['features'])
-        data_shapes = [('data', (batch_size, win, num_features))]
-
-        _pred_model.bind(data_shapes=data_shapes, label_shapes=None,
-                         for_training=False)
-        arg_params = _mxnet_utils.params_from_dict(state['_pred_model']['arg_params'])
-        aux_params = _mxnet_utils.params_from_dict(state['_pred_model']['aux_params'])
-        _pred_model.init_params(arg_params=arg_params, aux_params=aux_params)
-        state['_pred_model'] = _pred_model
-
-        return ActivityClassifier(state)
-
-    @classmethod
-    def _native_name(cls):
-        return "activity_classifier"
-
-    def _get_version(self):
-        return self._PYTHON_ACTIVITY_CLASSIFIER_VERSION
-
-    def export_coreml(self, filename):
+if use_cpp:
```

Comment:
@TobyRoseman Is this an okay thing to do? i.e define a class inside an if

Reply:
Well I'm not finding anything about this specifically in PEP8, but would recommend not doing it.

It adds a lot of additional nesting. Also it really break the diff; I can't tell what has changed and what hasn't.

Reply:
@TobyRoseman What is an alternative to the if-else here? We want to keep the class names the same and use one or the other based on what codepath we want to select. 
PS This is a temporary hack to enable using both Python and C++ implementations till we remove Python CustomModel class completely.

Reply:
Why do you need to keep the class names the same? Python is dynamically typed so I don't think it should matter. If you do really need to keep the classnames the same, I'm not sure there is another option.

Reply:
Per offline discussion, I think we should find a way to get the flag behavior testable without changing any names. The goal is to make both implementations testable without code changes, not to perform the actual migration (throwing the switch)...